File: Qualification Round 2009/C.py
#!/usr/bin/python
#
# Problem: Welcome to Code Jam
# Language: Python
# Author: KirarinSnow
# Usage: python thisfile.py <input.in >output.out


def compute():
    wc = 'welcome to code jam'
    counter = map(lambda x: [x, 0], wc)
    s = raw_input()

    for c in s:
        if c == 'w':
            counter[0][1] += 1
        for i in range(1, len(counter)):
            if c == counter[i][0]:
                counter[i][1] += counter[i-1][1]
                counter[i][1] %= 10000
    return counter[-1][1]

for i in range(input()):
    print "Case #%d: %04d" % (i+1, compute())

File: Qualification Round 2009/C.rb
#!/usr/bin/ruby
#
# Problem: Welcome to Code Jam
# Language: Ruby
# Author: KirarinSnow
# Usage: ruby thisfile.rb <input.in >output.out


def compute()
    $chars = readline.to_a[0]

    $ct = [0] * ($wc.length + 1)
    $ct[0] = 1
    
    for $j in 0 ... $chars.length
        $ch = $chars[$j]
	for $k in 0 ... $wc.length
            if $wc[$k] == $ch
                $ct[$k+1] = ($ct[$k] + $ct[$k+1])
                $ct[$k+1] %= 10000
            end
        end
    end

    return $ct[$wc.length]
end

$wc = 'welcome to code jam'.to_a[0]

for $i in 1 .. readline.to_i
    print "Case \##$i: %04d\n" % compute()
end
